D4Companion.Services/ScreenProcessHandler.cs

```diff
@@ -130,867 +113,514 @@ private void initImageList()
-            currentScreenTooltipFilter = currentScreenTooltipFilter.ThresholdBinaryInv(new Gray(_settingsManager.Settings.ThresholdMin), new Gray(_settingsManager.Settings.ThresholdMax));
+            var itemAffixLocationBag = new ConcurrentBag<ItemAffixLocationDescriptor>();
 
-            //// Clone image for GUI
-            //var currentScreenTooltipGui = _currentScreenTooltip.Clone();
-
-            // Create image for GUI
-            var currentScreenTooltipGui = new Image<Bgr, byte>(currentScreenTooltipFilter.Width, currentScreenTooltipFilter.Height, new Bgr());
-            currentScreenTooltipGui = currentScreenTooltipFilter.Convert<Bgr, byte>();
-
-            ConcurrentBag<List<ItemAffixLocationDescriptor>> itemAffixLocationBag = new ConcurrentBag<List<ItemAffixLocationDescriptor>>();
             Parallel.ForEach(_imageListItemAffixLocations.Keys, itemAffixLocation =>
             {
-                try
-                {
-                    itemAffixLocationBag.Add(FindItemAffixLocation(currentScreenTooltipFilter, Path.GetFileNameWithoutExtension(itemAffixLocation)));
-                }
-                catch (Exception exception)
-                {
-                    _logger.LogError(exception, $"{MethodBase.GetCurrentMethod()?.Name}");
-                }
+                var foundLocations = FindItemAffixLocation(_currentScreenTooltipBin, Path.GetFileNameWithoutExtension(itemAffixLocation));
+
+                if (foundLocations != null) foundLocations.ForEach(itemAffixLocationBag.Add);
             });
 
-            // Combine results
-            var itemAffixLocations = new List<ItemAffixLocationDescriptor>();
-            foreach ( var affixLocation in itemAffixLocationBag )
-            {
-                itemAffixLocations.AddRange(affixLocation);
-            }
+            var processedScreen = _currentScreenTooltipBin;
 
-            foreach (var itemAffixLocation in itemAffixLocations)
+            if (itemAffixLocationBag.Any())
             {
-                _currentTooltip.ItemAffixLocations.Add(itemAffixLocation.Location);
+                var currentScreenTooltipGui = _currentScreenTooltipBin.Clone();
+                _currentTooltip.ItemAffixLocations.Clear();
 
-                Point[] points = new Point[]
+                foreach (var itemAffixLocation in itemAffixLocationBag)
                 {
-                    new Point(itemAffixLocation.Location.Left,itemAffixLocation.Location.Bottom),
-                    new Point(itemAffixLocation.Location.Right,itemAffixLocation.Location.Bottom),
-                    new Point(itemAffixLocation.Location.Right,itemAffixLocation.Location.Top),
-                    new Point(itemAffixLocation.Location.Left,itemAffixLocation.Location.Top)
-                };
+                    _currentTooltip.ItemAffixLocations.Add(itemAffixLocation.Location);
+                    DrawRect(currentScreenTooltipGui, itemAffixLocation.Location);
```

Comment:
See notes on line 200/205.
No need for an extra one-line function. The static `CvInvoke.Rectangle` is already OK.